docs/source/reference/pysupported.rst

```diff
@@ -796,6 +796,23 @@ range of possible failures. However, the dictionary can be safely read from
 multiple threads as long as the contents of the dictionary do not
 change during the parallel access.
 
+Dictionary comprehension
+''''''''''''''''''''''''
+
+Numba supports dictionary comprehension under the assumption that a
+``numba.typed.Dict`` instance can be created from the comprehension.  For
+example::
+
+  In [1]: from numba import njit
+
+  In [2]: @njit
+  ...: def foo(n):
+  ...:     return {i: i**2 for i in range(n)}
+  ...:
```

Comment:
```suggestion
In [2]: @njit
   ...: def foo(n):
   ...:     return {i: i**2 for i in range(n)}
   ...:
```

Reply:
well spotted, colons are lined up in https://github.com/numba/numba/pull/6736/commits/7145e6cdf6f9d53a93281d89c7efe66f871b3702

Reply:
excellent, ready to smoketest then?

Reply:
Think so!